docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -12,30 +12,75 @@ ConnectorFactory
-following services:
+plugin. The connector factory is a simple interface responsible for providing
+the connector name and creating an instance of a ``Connector`` object.
+A basic connector implementation that only supports reading, but
+not writing data, should return instances of the following services:
 
-* ``ConnectorMetadata``
-* ``ConnectorSplitManager``
-* ``ConnectorRecordSetProvider``
+* :ref:`connector-metadata`
+* :ref:`connector-split-manager`
+* :ref:`connector-record-set-provider` or :ref:`connector-page-source-provider`
+
+.. _connector-metadata:
 
 ConnectorMetadata
 ^^^^^^^^^^^^^^^^^
 
-The connector metadata interface has a large number of important
-methods that are responsible for allowing Trino to look at lists of
-schemas, lists of tables, lists of columns, and other metadata about a
-particular data source.
+The connector metadata interface allows Trino to get a lists of schemas,
+tables, columns, and other metadata about a particular data source.
+
+A basic read-only connector should implement the following methods:
+
+* ``listSchemaNames``
+* ``listTables``
+* ``streamTableColumns``
+* ``getTableHandle``
+* ``getTableMetadata``
+* ``getColumnHandles``
+* ``getColumnMetadata``
+
+If you are interested in seeing strategies for implementing more methods,
+look at the :doc:`example-http` and the Cassandra connector. If your underlying
+data source supports schemas, tables and columns, this interface should be
+straightforward to implement. If you are attempting to adapt something that
+is not a relational database (as the Example HTTP connector does), you may
+need to get creative about how you map your data source to Trino's schema,
+table, and column concepts.
+
+The connector metadata interface allows to also implement other connector
+features, like:
+
+* Schema management, that is creating, altering and dropping schemas, tables,
+  table columns, views, and materialized views.
+* Support for table and column comments, and properties.
+* Schema, table and view authorization.
+* Executing :doc:`table-functions`.
+* Providing table statistics used by the CBO and collecting statistics
+  during writes and when analyzing selected tables.
+* Data modification, that is:
+
+  * inserting, updating, and deleting rows in tables,
+  * refreshing materialized views,
+  * truncating whole tables,
+  * and creating tables from query results.
 
-This interface is too big to list in this documentation, but if you
-are interested in seeing strategies for implementing these methods,
-look at the :doc:`example-http` and the Cassandra connector. If
-your underlying data source supports schemas, tables and columns, this
-interface should be straightforward to implement. If you are attempting
-to adapt something that is not a relational database (as the Example HTTP
-connector does), you may need to get creative about how you map your
-data source to Trino's schema, table, and column concepts.
+* Role and grant management.
+* Pushing down:
+
```

Comment:
Can also mention Dynamic filter here as that is a type of pushdown that needs to be explicitly implemented by the connector

Reply:
This section is specifically about `ConnectorMetadata`. I'd describe dynamic filtering in a follow-up PR and then I can reference it here.